### ISSUE TITLE
Serverless implicitly lowercases object suffix when using existing S3

### ISSUE THREAD

**Issue by notnami** [serverless/serverless#6409]
# This is a Bug Report

## Description

- What went wrong?

I'm setting an event on an existing S3 bucket, as described in https://serverless.com/framework/docs/providers/aws/events/s3#using-existing-buckets. When setting the event this way, `serverless` implicitly lowercases the object suffix in the event rule. This causes the event not to work when the created object has a suffix that is not lowercase.

- What did you expect should have happened?

Serverless should preserve the casing of the suffix when creating the event rule. For comparison, the [existing-s3](https://www.npmjs.com/package/serverless-plugin-existing-s3) plugin preserves the casing.

- What was the config you used?

```yaml
      - s3:
          bucket: ${opt:s3-bucket}
          events:
            - s3:ObjectCreated:*
          rules:
            - prefix: my-output/
            - suffix: _LATEST
          existing: true
```

Similar or dependent issues:

N/A

## Additional Data

- Serverless version: 1.48.0
- OS: Ubuntu 18.04

**Comment by nebur395** [comment]
Hi,

Unfortunately we are facing the same issue. We are trying to use that existing flag:  
```yml
customFunctionName:
    handler: src.handler
    events:
      - s3:
          bucket: ${self:custom.bucketName}
          events:
            - s3:ObjectCreated:*
          rules:
            - suffix: .txt
            - prefix: FolderInCamelCase/
          existing: true
```
After deploying our lambda, we have took a look at the generated template in our stack, and it still keeps the desired behaviour using CamelCase naming in prefix property. However, when we take a look at the lambda s3 event configuration, we don't know why, but it appears in lowercase:
```yml
Details:
  Event type: ObjectCreated
  Notification name: arnName
  Prefix: folderincamelcase/
  Suffix: .txt
```
It will be great to have some feedback soon, because for us is kinda blocking issue since S3 is case sensitive and that event is unable to trigger any action.

Thanks you a lot! 😄

**Comment by pmuens** [comment]
Thanks for opening @notnami and thanks for confirming @nebur395 :+1:

We're looking into that right now...

**Comment by pmuens** [comment]
Just a quick update that this should be fixed via #6456.
